test/e2e-olm/kibana_test.go

```diff
@@ -0,0 +1,95 @@
+package e2e
+
+import (
+    goctx "context"
+    "fmt"
+    "testing"
+
+    consolev1 "github.com/openshift/api/console/v1"
+    "github.com/openshift/elasticsearch-operator/pkg/constants"
+    "github.com/openshift/elasticsearch-operator/pkg/k8shandler/kibana"
+    "github.com/openshift/elasticsearch-operator/test/utils"
+    framework "github.com/operator-framework/operator-sdk/pkg/test"
+    "github.com/operator-framework/operator-sdk/pkg/test/e2eutil"
+    "k8s.io/apimachinery/pkg/types"
+)
+
+func TestKibana(t *testing.T) {
+    registerSchemes(t)
+    t.Run("kibana-group", func(t *testing.T) {
+        t.Run("KibanaDeployment", KibanaDeployment)
+    })
+}
+
+func KibanaDeployment(t *testing.T) {
+    ctx := framework.NewTestCtx(t)
+
+    namespace, err := ctx.GetNamespace()
+    if err != nil {
+        t.Fatal(err)
+    }
+    t.Logf("Found namespace: %v", namespace)
+
+    // get global framework variables
+    f := framework.Global
+    // wait for elasticsearch-operator to be ready
+    err = e2eutil.WaitForDeployment(t, f.KubeClient, namespace, "elasticsearch-operator", 1, retryInterval, timeout)
+    if err != nil {
+        t.Fatal(err)
+    }
+
+    if err = createOpenshiftLoggingNS(f, ctx); err != nil {
+        t.Fatal(err)
+    }
+
+    if err = createElasticsearchSecret(f, ctx, constants.OpenshiftNS); err != nil {
+        t.Fatal(err)
+    }
+
+    uuid := utils.GenerateUUID()
+    esDeploymentName := fmt.Sprintf("elasticsearch-cdm-%v-1", uuid)
+    _, err = createElasticsearchCR(f, ctx, constants.OpenshiftNS, uuid)
+    if err != nil {
+        t.Fatal(err)
+    }
+
+    err = e2eutil.WaitForDeployment(t, f.KubeClient, constants.OpenshiftNS, esDeploymentName, 1, retryInterval, timeout)
+    if err != nil {
+        t.Errorf("timed out waiting for Deployment %v: %v", esDeploymentName, err)
+    }
+
+    if err = createTrustedCABundle(f, ctx); err != nil {
```

Comment:
shouldn't be necessary to do this

Reply:
Actually, EO assumes a ConfigMap for the TrustedCABundle when deploying Kibana or else the following error is popping up:
```
{"level":"error","ts":1586172219.0207708,"logger":"kubebuilder.controller","msg":"Reconciler error","controller":"kibana-controller","request":"e2e-test-14374/kibana","error":" does not yet contain expected key ca-bundle.crt","stacktrace":"github.com/go-logr/zapr.(*zapLogger).Error\n\t/go/src/github.com/openshift/elasticsearch-operator/vendor/github.com/go-logr/zapr/zapr.go:128\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/go/src/github.com/openshift/elasticsearch-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:217\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1\n\t/go/src/github.com/openshift/elasticsearch-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:158\nk8s.io/apimachinery/pkg/util/wait.JitterUntil.func1\n\t/go/src/github.com/openshift/elasticsearch-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133\nk8s.io/apimachinery/pkg/util/wait.JitterUntil\n\t/go/src/github.com/openshift/elasticsearch-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:134\nk8s.io/apimachinery/pkg/util/wait.Until\n\t/go/src/github.com/openshift/elasticsearch-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88"}
```